raiden/api/python.py

```diff
@@ -70,14 +68,7 @@ def event_filter_for_payments(
             event.target == target_address
         )
     )
-    received_and_target_matches = (
-        isinstance(event, (EventPaymentReceivedSuccess)) and
-        (
-            target_address is None or
-            target_address == our_address
-        )
-    )
-    return sent_and_target_matches or received_and_target_matches
+    return sent_and_target_matches or isinstance(event, (EventPaymentReceivedSuccess))
```

Comment:
I think, if you're filtering for a `target`, as in the sense of a `partner`, instead of returning all `EventPaymentReceivedSuccess`, you should filter for events which `initiator` attribute is the `target_address` parameter

Reply:
So, IMO, the expected behavior would be changing the parameter name to `partner_address` and filtering for sent payments which `event.target == partner_address` or received payments which `event.initiator == partner_address`